### PR TITLE
Adds token invalidation when the server responds with 401 or 403

### PR DESCRIPTION
**Pull request by eluciano11**

### CODE REVIEW

File: server/mocks/api/token-refresh.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ module.exports = function(app) {
     jwt.verify(req.body.token, 'secret', function(err, decoded) {
       if (err) {
         res
-          .status(401)
+          .status(402)
```
Comment (jpadilla) [inline server/mocks/api/token-refresh.js]: why change this?

Comment (eluciano11) [inline server/mocks/api/token-refresh.js]: mmm typo on testing I guess...

File: addon/authenticators/jwt.js [hunk 2]
```diff
@@ -238,6 +238,7 @@ export default TokenAuthenticator.extend({
           }
         });
       }, (xhr, status, error) => {
+        this.handleTokenRefreshFail(xhr.status);
         Ember.Logger.warn(`Access token could not be refreshed - server responded with ${error}.`);
```
Comment (jpadilla) [inline addon/authenticators/jwt.js:242]: Small nitpick, let's log first and then do `handleTokenRefreshFail`

**Comment by basz** [comment]
Q: Why only for 401 and/or 403? According to the [specs](https://tools.ietf.org/html/rfc6749#section-5.2) 400 might be returned too when the refresh token has expired ('invalid_grant')? 

ps. This repo is NOT about the default oauth2 implementation in ember-simple-auth is it?

**Comment by jpadilla** [comment]
@basz right, this isn't about the oauth2 implementation in ESA. For now I'd rather have this on 401/403. @eluciano11 thoughts?

**Comment by jpadilla** [comment]
@eluciano11 we also need some docs on this.

**Comment by eluciano11** [comment]
Yeah, I decided to use 401 and 403 because this will be called on the token refresh. When a server returns a `Forbidden (403)` it mean it understood the request but it doesn't authorise it. When the client sends any other request with an unauthorized token the server will respond with `Unauthorized (401)`.

HTTP 400 is a `Bad Request` and I don't know if that is very common in this scenario. I've used Django Rest Framework with JWT and from my experience the backend usually returns a `403` when the token isn't refreshed.

If your backend returns a HTTP 400 when the token fails to refresh you can extend the JWT authenticator and redefine the `handleTokenRefreshFail` function.

**Comment by basz** [comment]
400 is in the oauth2 spec. 

Extending is acceptable though

Thx

**Pull request merged**